### DISCUSSION TITLE
Dompdf generating corrupt file :( Need help!!!

### DISCUSSION THREAD

**Discussion by DaniLgza** [dompdf/dompdf#2804] [Show and tell]
![image](https://user-images.githubusercontent.com/38362794/158591322-9be8ccf1-3895-4d2e-9daf-158f6e74e4e8.png)

![image](https://user-images.githubusercontent.com/38362794/158591478-8b081db4-05f7-470e-9944-ecd0567e37c4.png)

I'm trying to generate a very simple pdf with Dompdf, but its not working. The PDF is generated, but i could not open in the browser and neither with Adobe PDF. 

If i try to open it with NotePad+, the file contains the next:

`<html> <head>    <meta charset="utf-8">    <meta http-equiv="X-UA-Compatible" content="IE=edge">    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">    <meta name="description" content="ads">    <meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate" />    <meta http-equiv="Pragma" content="no-cache" />    <meta http-equiv="Pragma-directive" content="0" />    <meta http-equiv="Cache-directive" content="no-cache" />    <meta http-equiv="Expires" content="0" />    <title> > ads</title>    <link rel="shortcut icon" type="image/x-icon" href="/img/favicon.ico"/>    <link rel='dns-prefetch' href='https://fonts.googleapis.com' />    <link rel="stylesheet" href="/css/ads/bootstrap.min.css?v=5">      <link rel="stylesheet" href="/css/backoffice/ats.css?v=10">    <link rel="stylesheet" href="/css/ads/admin.css?v=1">        <link href="/vendor/fortawesome/font-awesome/css/all.min.css" rel="stylesheet" type="text/css">    <link href="https://fonts.googleapis.com/css?family=Nunito:300,400,500,700&display=swap" rel`

Its seems, its not loading the html I wrote "Hello world".

I've also tried to use ob_end_clean;, but also not works.

What i'm doing wrong? Could anyone help me please!

**Comment by bsweeney** [comment]
Either something is intercepting the output and wrapping it in HTML, or output buffering is enabled and some HTML content was generated prior to using Dompdf. If you're running in the context of a framework you should look to see how you can control the output so that the extra content isn't generated when you're using this method. Otherwise, uncomment `ob_end_clean()` in your snippet and add an `exit` statement after you stream the PDF.

**Reply by DaniLgza** [reply]
Thanks a lot @bsweeney, adding `exit` statement works!!! 🙌
